### PR TITLE
[FIX] purchase_stock : prevent wrong destination with orderpoint

### PR DESCRIPTION
**Pull request by fmdl**
Description of the issue/feature this PR addresses:
- create 2 warehouses A & B
- clic Automated Order in replenishment on product in for warehouse A
- open PO
- change picking type to the other warehouse B
- confirm order and picking.
--> product is now A, it should be B

The issue can also appear after duplicate.

@amoyaux @simongoffin 

Note : I dont know if it is the best fix.

https://user-images.githubusercontent.com/16716992/124234227-0d115280-db14-11eb-9f7e-551958e87dd2.mov

--
I confirm I have signed the CLA and read the PR guidelines at www.odoo.com/submit-pr

### CODE REVIEW

**Comment by fmdl** [comment]
OPW #2588082

File: addons/purchase_stock/models/purchase.py [hunk 1]
```diff
@@ -441,9 +441,21 @@ def _prepare_stock_moves(self, picking):
             extra_move_vals['move_dest_ids'] = False  # don't attach
             res.append(extra_move_vals)
         return res
+    
+    def _check_orderpoint_picking_type(self):
+        warehouse_loc = self.order_id.picking_type_id.warehouse_id.view_location_id
+        if self.orderpoint_id.location_id:
```
Comment (Whenrow) [inline addons/purchase_stock/models/purchase.py:447]: I would prefer using the parent_path to check if the location is below the warehouse
``` python
if not warehouse_loc.parent_path in self.orderpoint.location_id.parent_path:
    raise ...
```

Also a quick test to check the error would be nice. I guess amending an existing test in odoo/addons/purchase_stock/tests/test_reordering_rule.py is good.

**Comment by DjamelTouati** [comment]
Done in: https://github.com/odoo/odoo/pull/75306

**Comment by fmdl** [comment]
Close by https://github.com/odoo/odoo/pull/75306

**Pull request closed without merging**